### PR TITLE
Scale hide value label

### PR DESCRIPTION
**Pull request by rugheid**
This pull request adds an optional boolean to all scale answer formats to hide the value label of the scale. The boolean value is not included in the initialisers, but has to be set explicitly if one wants to override the default behaviour. Two steps are added to ORKCatalog to the scale task. The first added step contains a discrete horizontal scale with a hidden value label. The second added step contains a continuous vertical scale with a hidden value label.

### CODE REVIEW

**Comment by rsanchezsaez** [comment]
I have not reviewed the code, but I suggest you move the new scale test cases from `ORKCatalog` to `ORKTest`.

The general idea behind `ORKCatalog` is to keep it a light and simple catalog of the different question types, rather than an exhaustive one of all their available options.

**Comment by rugheid** [comment]
Ok, I understand! I'll move it.

**Comment by rugheid** [comment]
I removed the extra steps from ORKCatalog and I think there isn't actually anything to add to te tests. So I guess this pull request is ready for review.

**Comment by YuanZhu-apple** [comment]
@rugheid Thanks for this PR!

Can you give some use cases of this feature? Do you have concerns about current value shown to users could affect their answer?

**Comment by rugheid** [comment]
@YuanZhu-apple I worked with a couple of doctors on my last project. They all agreed that showing the user a number would affect the results. When a user just sees a bar, he/she will have to choose depending on his/her feeling. The doctors thought that otherwise the user would start thinking in numbers and just set the bar to the number he/she had in mind.
This applies best to the continuous scale, but I can imagine some developers to remove the label for the discrete scale and text scale as well.

File: ResearchKit/Common/ORKAnswerFormat.h [hunk 1]
```diff
@@ -477,6 +482,11 @@ ORK_CLASS_AVAILABLE
  */
 @property (strong, nullable) UIImage *minimumImage;
 
+/**
+ A Boolean value indicating whether the value label should be hidden. (read-only)
+ */
+@property (assign, getter=shouldHideValueLabel) BOOL hideValueLabel;
```
Comment (YuanZhu-apple) [inline ResearchKit/Common/ORKAnswerFormat.h]: Would `hideSelectedValue` be a better name?

File: ResearchKit/Common/ORKScaleSliderView.m [hunk 2]
```diff
@@ -423,6 +431,14 @@ - (instancetype)initWithFormatProvider:(id<ORKScaleAnswerFormatProvider>)formatP
                                                      options:NSLayoutFormatAlignAllCenterY | NSLayoutFormatDirectionLeftToRight
                                                      metrics:@{@"kMargin": @(kMargin)}
                                                        views:views]];
+            
+            if ([_formatProvider shouldHideValueLabel]) {
+                [self addConstraints:
+                 [NSLayoutConstraint constraintsWithVisualFormat:@"V:[_valueLabel(==0)]"
```
Comment (YuanZhu-apple) [inline ResearchKit/Common/ORKScaleSliderView.m]: I have seen this constraint twice, can you consolidate them? Maybe merge the latest changes from master branch first.

**Comment by YuanZhu-apple** [comment]
@rugheid Your point is valid. Thanks for the explanation.

**Comment by YuanZhu-apple** [comment]
Can you also update `ORKESerialization`, which handles the JSON encoding/decoding.

**Comment by rugheid** [comment]
@YuanZhu-apple I changed the name of the property to `hideSelectedValue`, updated `ORKESerialization` and changed `ORKScaleSliderView` to use the constraints only once. All changes from master were already merged.

**Comment by YuanZhu-apple** [comment]
Thanks! 
Can you also resolve the conflicts?

**Comment by umerkhan-apple** [comment]
@rugheid Were you able to make time to resolve the conflicts?

**Comment by rugheid** [comment]
@umerkhan-apple I think I resolved most conflicts (not yet pushed though). I did not have time yet to test properly. I'll try to get it done this weekend.

**Comment by umerkhan-apple** [comment]
@rugheid Awesome, sounds good.

**Comment by umerkhan-apple** [comment]
@rugheid Any updates on this PR?

**Comment by rugheid** [comment]
@umerkhan-apple I did not find the time yet to update it, sorry...

**Comment by umerkhan-apple** [comment]
@rugheid No worries, let me know if you need any help. Also let me know whenever you've update it, so I can do a final review.

**Comment by rugheid** [comment]
@umerkhan-apple Will do! Thanks

**Comment by pmitche** [comment]
What's the status of this PR?

**Comment by rugheid** [comment]
I've not been able to fix the conflicts yet. If you don't want to keep this pull request open, I'll open a new one once I've fixed the conflicts.
Sorry for taking so long.

**Comment by YuanZhu-apple** [comment]
ok, closed for now.

**Pull request closed without merging**